pkg/discovery/discovery.go

```diff
@@ -226,3 +235,102 @@ func (d *tidbDiscovery) DiscoverDM(advertisePeerUrl string) (string, error) {
 	delete(currentCluster.peers, podName)
 	return fmt.Sprintf("--join=%s", strings.Join(mastersArr, ",")), nil
 }
+
+func (d *tidbDiscovery) VerifyPDEndpoint(pdURL string) (string, error) {
+	// save a copy of pdURL for failure
+	copyPDEndpoint := pdURL
+	pdEndpoint := parsePDURL(pdURL)
+
+	ns := os.Getenv("MY_POD_NAMESPACE")
+	tc, err := d.cli.PingcapV1alpha1().TidbClusters(ns).Get(pdEndpoint.tcName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get the tidbcluster when verifying PD endpoint")
+		return pdURL, err
+	}
+
+	noScheme := false
+	if len(pdEndpoint.scheme) == 0 {
+		noScheme = true
+		if tc.IsTLSClusterEnabled() {
+			pdEndpoint.scheme = "https"
+		} else {
+			pdEndpoint.scheme = "http"
+		}
+		pdURL = fmt.Sprintf("%s://%s", pdEndpoint.scheme, pdURL)
+	}
+
+	if d.pdEndpointHealthCheck(tc, pdURL, pdEndpoint.pdMemberName) {
```

Comment:
No need to verify the health of the pdURL, just return it and the healthy peer members if any.